src/dlgRoomExits.cpp

```diff
@@ -312,7 +312,7 @@ void dlgRoomExits::save()
         pR->setSpecialExit(-1, value);
     }
 
-    if (nw->isEnabled() && nw->text().size() > 0 && mpHost->mpMap->mpRoomDB->getRoom(nw->text().toInt()) != 0) {
+    if (nw->isEnabled() && nw->text().size() > 0 && mpHost->mpMap->mpRoomDB->getRoom(nw->text().toInt()) != nullptr) {
```

Comment:
Actually XX->text().size() can never be negative so this and following repeats can be shortened to:

    if (nw->isEnabled() && nw->text().size() && mpHost->mpMap->mpRoomDB->getRoom(nw->text().toInt())) {